### PR TITLE
Array: remove extraneous code from RandomState

### PR DESCRIPTION
**Pull request by gjoseph92**
There were a few lines hanging around that weren't doing anything, probably left over from previous refactors.

- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: dask/array/random.py [hunk 1]
```diff
@@ -112,7 +112,6 @@ def _broadcast_any(ar, shape, chunks):
         # Broadcast all arguments, get tiny versions as well
         # Start adding the relevant bits to the graph
         dsk = {}
-        dsks = []
```
Comment (gjoseph92) [inline dask/array/random.py:115]: This is unused (besides appending things to it)

File: dask/array/random.py [hunk 2]
```diff
@@ -165,7 +162,6 @@ def _broadcast_any(ar, shape, chunks):
                     arg.append(ar)
                 else:
                     if isinstance(ar, Array):
-                        dependencies.append(ar)
```
Comment (gjoseph92) [inline dask/array/random.py:168]: Adding the full `ar` to `dependencies` (in a loop over every block) is unnecessary. We've already added the broadcast+rechunked array to dependencies [here](https://github.com/dask/dask/pull/7487/files#diff-bacbead89e1b64dbdf54cbe8878e9a52445945d2c2710b2a11edc65364c6b16dR122). This non-broadcast version isn't directly referenced (though it's indirectly referenced by the broadcast+rechunked version), so it's just bloat in the HLG.

**Comment by jsignell** [comment]
Thanks for opening this @gjoseph92! It looks good to me, but given the nature of the PR I'd prefer if tests were passing. Do you mind merging/rebasind off main to trigger them again?

**Comment by gjoseph92** [comment]
@jsignell all green besides an unrelated windows test!

**Pull request merged**